packages/application-components/src/utils/default-form-buttons.tsx

```diff
@@ -27,20 +27,21 @@ const primaryDefaultProps: Pick<
   dataAttributes: {},
 };
 
-const getFormattedLabel = (label: Label, intl: IntlShape) =>
-  typeof label === 'string' ? label : intl.formatMessage(label);
-
-const FormPrimaryButton = (props: Props) => {
+const useFormattedLabel = (label: Label) => {
   const intl = useIntl();
-  return (
-    <PrimaryButton
-      label={getFormattedLabel(props.label, intl)}
-      onClick={props.onClick}
-      isDisabled={props.isDisabled}
-      {...filterDataAttributes(props.dataAttributes)}
-    />
-  );
+
+  return typeof label === 'string' ? label : intl.formatMessage(label);
 };
+
+const FormPrimaryButton = (props: Props) => (
+  <PrimaryButton
+    label={useFormattedLabel(props.label)}
```

Comment:
Careful. You might want to move this to L37 and store it in a var cause if we ever conditionally render the order of hooks can change easily.